src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java

```diff
@@ -39,10 +39,7 @@
  * int x = (y / 2 + 1); // parens around assignment rhs
  * for (int i = (0); i &lt; 10; i++) {  // parens around literal
  * t -= (z + 1);                     // parens around assignment rhs
- * boolean a = (x &gt; 7 &amp;&amp; y &gt; 5)      // parens around expression
- *             || z &lt; 9;
- * boolean b = (~a) &gt; -27            // parens around ~a
- *             &amp;&amp; (a-- &lt; 30);        // parens around expression
+ * boolean b = (~a) &gt; -27;            // parens around ~a
```

Comment:
Cases removed do not fall under the convention we said we would target. https://github.com/checkstyle/checkstyle/issues/9957#issuecomment-829744968

Why were these removed?